back-end/app.js

```diff
@@ -124,4 +135,90 @@ app.get('/shows-trakt', (req, res, next) => {
   }
 })
 
+app.get('/shows-trakt/:id', (req, res, next) => {
+  const badRequestError = {
+   status: 400,
+   error:"Bad Request - request couldn't be parsed",
+   message: "Content type (show or movie) is not indicated e.g. /shows-trakt/1390/?type=movie",
+   path: "/shows-trakt/:id",
+  }
+  const notFoundError = {
+    status: 404,
+    error:"Not Found - method exists, but no record found",
+    message: "Content with the indicated Trakt id is not found",
+    path: "/shows-trakt/:id"
+  }
+  let traktURL, tmdbType; 
+  //Bad Request error if content type is not given
+  if (Object.keys(req.query).length === 0){
+    res.json(badRequestError);
+  //Otherwise return extended show information 
+  }else if (req.query.type=='show'){
+    traktURL=`https://api.trakt.tv/shows/${req.params.id}`;
+    tmdbType='tv';
+  }else if (req.query.type=='movie'){
+    traktURL=`https://api.trakt.tv/movies/${req.params.id}`;
+    tmdbType='movie';
+  }else //return Bad Request error for other errors
+    res.json(badRequestError);
+  return axios
```

Comment:
Would it be possible to put some sort of check before the return statement/axios call (the first of which isn't really necessary for this case, I think?). When we want to return a `badRequestError`, the program still ends up hitting the axios call and tries to send more information from there, I think.
Doing another if statement after this first one (where `traktURL` and `tmdbType` are assigned) may take care of it. So something like this maybe:
```
if (tmdbType !== undefined) {
 put the axios stuff here
}
```
![image](https://user-images.githubusercontent.com/35475933/113710545-d90fc800-96b1-11eb-877b-304f54460973.png)

Reply:
yup, just added if statement, thanks